Bottom: 36a12fe5001a4ce3c272d3eab395600bead6f1bf
Top:    9d98b040ab3f637cacece4a2a4f16cdc67a04d3b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-09 13:11:16 +0100

Refresh of layer-fixes

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index e45de03..bd38db8 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1402,7 +1402,7 @@ DrawLayerGroup (int group, const BoxType * screen)
         strcmp (Layer->Name, "route") == 0)
       rv = 0;
 
-    if (layernum < max_layer /*&& Layer->On*/) {
+    if (layernum < max_layer && Layer->On) {
 
       if (!first_run)
         gui->set_layer (0, group, 0);
@@ -1582,14 +1582,12 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     // Draw in numerical order when in 3D view
     group = global_view_2d ? GetLayerGroupNumberByNumber (LayerStack[i]) : orderi;
 
-//    if (!do_group[group]) {
-    if (l->On && !do_group[group]) {
+    if (!do_group[group]) {
       do_group[group] = 1;
       drawn_groups[ngroups++] = group;
     }
   }
 
-
   /*
    * first draw all 'invisible' stuff
    */
